optuna/visualization/_terminator_improvement.py

```diff
@@ -40,6 +40,54 @@ def plot_terminator_improvement(
     error_evaluator: BaseErrorEvaluator | None = None,
     min_n_trials: int = DEFAULT_MIN_N_TRIALS,
 ) -> "go.Figure":
+    """Plot the potentials for future objective improvement.
+
+    This function visualizes the objective improvement potentials, evaluated
+    with `improvement_evaluator`.
+    It helps to determine whether we should continue the optimization or not.
+    You can also plot the error evaluated with
+    `error_evaluator` if the `plot_error` argument is set to :obj:`True`.
+    Note that this function may take some time to compute
+    the improvement potentials.
+
+    Example:
+
+        The following code snippet shows how to plot improvement potentials.
+
+        .. plotly::
+
+            import optuna
+
+            def objective(trial):
+                x = trial.suggest_float("x", -100, 100)
+                y = trial.suggest_categorical("y", [-1, 0, 1])
+                return x ** 2 + y
+
+            sampler = optuna.samplers.TPESampler(seed=10)
+            study = optuna.create_study(sampler=sampler)
+            study.optimize(objective, n_trials=10)
+
+            fig = optuna.visualization.plot_terminator_improvement(study)
+            fig.show()
```

Comment:
The example does not fully use the functionality of `plot_terminator_improvement()`. How about using the example from https://github.com/optuna/optuna/pull/4609#issuecomment-1541203793.
```suggestion
            from lightgbm import LGBMClassifier
            from sklearn.datasets import load_wine
            from sklearn.model_selection import cross_val_score
            from sklearn.model_selection import KFold

            import optuna
            from optuna.terminator import report_cross_validation_scores
            from optuna.visualization import plot_terminator_improvement


            def objective(trial):
                X, y = load_wine(return_X_y=True)

                clf = LGBMClassifier(
                    reg_alpha=trial.suggest_float("reg_alpha", 1e-8, 10.0, log=True),
                    reg_lambda=trial.suggest_float("reg_lambda", 1e-8, 10.0, log=True),
                    num_leaves=trial.suggest_int("num_leaves", 2, 256),
                    colsample_bytree=trial.suggest_float("colsample_bytree", 0.4, 1.0),
                    subsample=trial.suggest_float("subsample", 0.4, 1.0),
                    subsample_freq=trial.suggest_int("subsample_freq", 1, 7),
                    min_child_samples=trial.suggest_int("min_child_samples", 5, 100),
                )

                scores = cross_val_score(clf, X, y, cv=KFold(n_splits=5, shuffle=True))
                report_cross_validation_scores(trial, scores)
                return scores.mean()


            study = optuna.create_study()
            study.optimize(objective, n_trials=30)

            fig = plot_terminator_improvement(study, plot_error=True)
            fig.show()
```

Reply:
Good catch!
Thanks for your careful review!

Reply:
I think you need to merge master branch for this update

Reply:
I applied your comment. PTAL.

Reply:
Good catch! I had the same suggestion in mind, but was wondering if it was a little too complicated to write in the docstring. If it is no problem in this viewpoint, I completely agree this change.

Reply:
Maybe we could remove some unimportant parameters.